### PR TITLE
Check for the line hit for fall through branch sides

### LINKED ISSUE #441: Provide a way to check line hit when coverage type is not file
The function is its current implementation does not do what it is supposed to for C/C++ coverage.

More specifically `self._cov_type` is not set and `self.file_map` is not populated for such coverages to be used here.
I can take a pass on it on Monday.

### PR DESCRIPTION
**Pull request by Navidem**
In a C/C++ coverage report the line info is categorized based on the function name. This PR checks if a line in a given function is hit or not.

Fixes #441

### CODE REVIEW

File: src/fuzz_introspector/code_coverage.py [hunk 1]
```diff
@@ -221,6 +235,7 @@ def load_llvm_coverage(
                 found_name = target_name
 
     cp = CoverageProfile()
+    cp.set_type("file")
```
Comment (DavidKorczynski) [inline src/fuzz_introspector/code_coverage.py:238]: This should be "function" or something of this nature, but not "file", right?

Comment (Navidem) [reply src/fuzz_introspector/code_coverage.py:238]: Oops my bad, this was intended to be removed. Thanks for pointing out.

Btw, `function` type sounds good. I will use it.

Comment (DavidKorczynski) [reply src/fuzz_introspector/code_coverage.py]: awesome!

**Review by DavidKorczynski — APPROVED** [review]
lgtm!

Down the line I think it does make sense to make the code coverage API cleaner as described here https://github.com/ossf/fuzz-introspector/issues/441#issuecomment-1207196169 to avoid too many `if` statements across the code. This could be enforced if we require that the coverage type is specified for each `CoverageProfile`.

**Comment by Navidem** [comment]
> This could be enforced if we require that the coverage type is specified for each `CoverageProfile`.

+1 on enforcing coverage type in `CoverageProfile`. This will help as we move forwards with different langs.

**Pull request merged**